backend/functions/src/controllers/flashcard.js

```diff
@@ -90,3 +90,96 @@ exports.addFlashcardSet = functions.https.onCall(async (data, context) => {
     throw new functions.https.HttpsError("unknown", err.message, err);
   }
 });
+
+/**
+ * cloud function to update a flashcard set.
+
+  Expected input:
+    - "flashcardId"
+    - "title" <string (1-30 characters)>
+    - "category" <string (1-30 characters)>
+      - "cards" []
+      - "question" <string (1+ characters)>
+      - "answer" <string (1+ characters)>
+ */
+exports.updateFlashcardSet = functions.https.onCall(async (data, context) => {
+  if (!context.auth) {
+    const errMsg = "You must be authenticated to update a flashcard set.";
+    throw new functions.https.HttpsError("unauthenticated", errMsg);
+  }
+
+  // Make sure required fields are provided
+  const requiredFields = ["flashcardId", "title", "category", "cards"];
+  requiredFields.forEach((field) => {
+    if (!data[field]) {
+      const errMsg = `The "${field}" field must be provided.`;
+      throw new functions.https.HttpsError("invalid-argument", errMsg);
+    }
+  });
+
+  // Valdiate each field
+  if (!myUtils.isStrBetween(data.title, 1, 30)) {
+    throw new functions.https.HttpsError(
+        "invalid-argument",
+        "The \"title\" field must be a string between 1-30 characters.",
+    );
+  }
+  if (!myUtils.isStrBetween(data.category, 1, 30)) {
+    throw new functions.https.HttpsError(
+        "invalid-argument",
+        "The \"category\" field must be a string between 1-30 characters.",
+    );
+  }
+  if (
+    !Array.isArray(data.cards) ||
+    data.cards.length === 0 ||
+    !myUtils.arrayOfObjContainKeys(data.cards, ["question", "answer"])
+  ) {
+    const errMsg =
+      "The \"cards\" field must be a non-empty array of objects with keys" +
+      " \"question\" and \"answer\" with non-empty string values.";
+    throw new functions.https.HttpsError("invalid-argument", errMsg);
+  }
+
+  // Format data to be updated in the database
+  const flashcardSetMetaData = {
+    title: data.title.trim(),
+    category: data.category.trim(),
+    timestamp: Date.now(),
+  };
+  const flashcardSet = {
+    creatorId: context.auth.uid,
+    cards: data.cards.map((obj) => ({
```

Comment:
Probably should remove the `creatorId: context.auth.uid,` since the flashcard set creator id won't change (the Firebase `update()` function will only update the fields provided).

Reply:
I think i was failing the tests when it wasn't there. But now I have the same function for both add and update.

Reply:
Since we remove the `creatorId` field; it won't appear in the return object so the `expect(res.creatorId).toBe(context.auth.uid);` will end up failing